app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/PreferencesFragment.java

```diff
@@ -62,14 +90,80 @@ public boolean onPreferenceClick(Preference preference) {
                 Intent contactIntent = new Intent(Intent.ACTION_SENDTO);
                 contactIntent.setData(Uri.parse(getString(R.string.off_mail)));
                 contactIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
-                try{
+                try {
                     startActivity(contactIntent);
-                }catch (android.content.ActivityNotFoundException e){
+                } catch (android.content.ActivityNotFoundException e) {
 
-                    Toast.makeText(getActivity(), R.string.email_not_found,Toast.LENGTH_SHORT).show();
+                    Toast.makeText(getActivity(), R.string.email_not_found, Toast.LENGTH_SHORT).show();
                 }
                 return true;
             }
         });
     }
+
+    private class GetAdditives extends AsyncTask<Void, Integer, Boolean> {
+
+        private static final String ADDITIVE_IMPORT = "ADDITIVE_IMPORT";
+        private LoadToast lt = new LoadToast(getActivity());
+
+
+        @Override
+        protected void onPreExecute() {
+            super.onPreExecute();
+            lt.setText(getActivity().getString(R.string.toast_retrieving));
+            lt.setBackgroundColor(ContextCompat.getColor(getActivity(), R.color.blue));
+            lt.setTextColor(ContextCompat.getColor(getActivity(), R.color.white));
+            lt.show();
+        }
+
+        @Override
+        protected Boolean doInBackground(Void... arg0) {
+            boolean result = true;
+            Context context = getActivity();
+
+            String additivesFile;
+            switch (LocaleHelper.getLanguage(context)) {
+                case "fr":
```

Comment:
Either doing a very long case system with all the cases, or concatenate languages, and handling languages with non standard codes.

Reply:
yeah, I thought an approach before to do something like this:
```java
String language = LocaleHelper.getLanguage(SplashActivity.this);
            String additivesFile = "additives_en.json";
            additivesFile.replace("en", language);
```
but I'm unsure about having translation in every language in server at present.

Reply:
But if the user has changed the language and the selected language doesn't contain the translation so in that case, additives would not be changed to the new language and remain as it is.
It seems quite approachable by the code I've shown.